src/backend/api/reservations.js

```diff
@@ -31,6 +31,9 @@ router.post("/", async (request, response) => {
         contact_name: body.contact_name,
         contact_email: body.contact_email
       });
+      await knex('meals')
+        .where({ id: body.meal_id })
+        .decrement({ max_reservations: body.number_of_guests })
```

Comment:
There's a database concept called Transactions. Transactions allow you to run multiple queries, and make sure that they either _all succeed_ or they _all fail_.

Above, you execute 2 queries (1 to insert a reservation and 1 to decrease the number of meals). What happens if something goes wrong though (your database dies, or your network/internet access dies) in between those 2 queries? You'd end up having 1 reservation inserted in the database, but the number of meals will not have been decreased. Ohoo, trouble.

If you wrap both those queries in a transaction, you can protect against stuff like this, and make sure they either both succeed, or none does.

Knex documentation for transactions: http://knexjs.org/guide/transactions.html
More explanations on how transactions work: https://sebhastian.com/mysql-transaction/